scanner/decisions/0010-separate-versioning-for-vuln-bundles.md

```diff
@@ -0,0 +1,82 @@
+
+To address that scenario, Scanner created one bundle per Stackrox release.  Each bundle is created with an Exporter running the same ClairCore version used to build the release, ensuring compatibility.  Bundles are served at `https://definitions.stackrox.io/v4/V/vulnerabilities.zip` where `V` will be a Stackrox release version such as "4.5.0".  Scanner uses its release version tagged during build time to determine which version to fetch from.  The versions to build bundles for are taken from a file named `scanner/updater/version/RELEASE_VERSION`.
+
+For its purpose, this approach needs to be revised.
+
+1. The vulnerability schema rarely changes, but maintaining multiple bundles is expensive.  It requires multiple GitHub workflow runs, which increases the chances of failure and pressure on the security sources' endpoints.
+2. It isn't easy to backport fixes and updates to previous releases.  This requires patch releases, which trigger the release workflow.  Improving the release process is noisy, convoluted, and complex.
+3. The offline mode's "latest bundle" needs to exist as a complex solution to help users transition to the versioned bundle.  It contains all versions in a single bundle.  But one bundle per release makes the latest bundle large and, most of the time, essentially carries the same data since the schema is the same.
+
+## Decision
+
+We will implement a versioning stream for vulnerability bundles, bumped when we determine that ClairCore's schema has changed.
+
+Initially, two version streams will exist (table 1 below):
+
+| Stream | Git Ref      | Description                                                        |
+|--------|--------------|--------------------------------------------------------------------|
+| dev    | heads/master | This stream tracks the latest vulnerability schema                 |
+| v1     | heads/4.4.5  | This stream tracks the schema used by the current Stackrox release |
+
+On the subsequent releases (for example,. `4.4.6`), one of the two will be carried out:
+
+1.  The schema hasn't changed, so v1 now tracks the new release (`heads/4.4.6`).
+2.  The schema has changed, so v1 stays at current release (`heads/4.4.5`), and a new version, v2, is created to track the next release (`heads/4.4.6`).
+
+### Building bundle version in Exporter
+
+The GitHub workflow will have a map with "bundle version" as keys, and "git reference" as values, similar to table 1 above.  This can be stored statically in the matrix definition.
+
+For each version in the mapping, a bundle will be built based on the specified git ref and stored at `https://definitions.stackrox.io/v4/<bundle-version>/vulnerabilities.zip`.
+
+In every release cut, the key specified in `scanner/VULNERABILITY_VERSION` will be updated in the mapping to the current release branch reference (e.g., `heads/4.4.6`).  Notice that if the version is the same, it will be updated.  If it's new, it will be added, and the previous version will stay.
```

Comment:
Is this a new file? What happens to the old one?

Reply:
It seems you're referring to `scanner/updater/version/RELEASE_VERSION`. Is that right? It will continue to be, but holding version mappings, instead of ACS releases for building bundles.